include/boost/geometry/algorithms/detail/overlay/copy_segment_point.hpp

```diff
@@ -66,15 +65,14 @@ struct copy_segment_point_range
         cview_type cview(range);
         rview_type view(cview);
 
-        typedef typename boost::range_iterator<rview_type>::type iterator;
-        geometry::ever_circling_iterator<iterator> it(boost::begin(view), boost::end(view),
-                    boost::begin(view) + seg_id.segment_index, true);
+        std::size_t const segment_count = boost::size(view) - 1;
 
-        for (signed_size_type i = 0; i < offset; ++i, ++it)
-        {
-        }
+        while (offset < 0) { offset += segment_count; }
+
+        signed_size_type const target = (seg_id.segment_index + offset) % segment_count;
```

Comment:
It looks like it should be possible to calculate that without any loops and maybe even with one modulo operation.
Can `-offset` be greater than `segment_count`?

Reply:
Yes, modulo is probably possible. Offset is usually 1 (I've never encountered higher values)

Reply:
Fixed :heavy_check_mark: